metadata-ingestion/src/datahub/ingestion/source/datahub/datahub_database_reader.py

```diff
@@ -147,6 +147,53 @@ def query(self) -> str:
             version
         """
 
+    def _get_rows(
+        self, from_createdon: datetime, stop_time: datetime
+    ) -> Iterable[Dict[str, Any]]:
+        # Cursor for MySql and PostgreSQL are by default client side cursor and we need to create server side cursor -> https://docs.sqlalchemy.org/en/20/core/connections.html#using-server-side-cursors-a-k-a-stream-results
+        # For MySQL, we need to use SSCursor to create server side cursor
+        # For PostgreSQL, we need to use stream_results=True to create server side cursor and need to be run in transaction
+        with self.engine.connect() as conn:
+            if self.engine.dialect.name == "postgresql":
+                with conn.begin():  # Transaction required for PostgreSQL server-side cursor
+                    conn = conn.execution_options(
+                        stream_results=True,
+                        yield_per=self.config.database_query_batch_size,
+                    )
+                    result = conn.execute(
+                        self.query,
+                        {
+                            "exclude_aspects": list(self.config.exclude_aspects),
+                            "since_createdon": from_createdon.strftime(DATETIME_FORMAT),
+                        },
+                    )
+                    for row in result:
+                        yield dict(row)
+            elif self.engine.dialect.name == "mysql":  # MySQL
+                import MySQLdb
+
+                with contextlib.closing(
+                    conn.connection.cursor(MySQLdb.cursors.SSCursor)
+                ) as cursor:
+                    logger.debug(f"Using Cursor type: {cursor.__class__.__name__}")
+                    cursor.execute(
+                        self.query,
+                        {
+                            "exclude_aspects": list(self.config.exclude_aspects),
+                            "since_createdon": from_createdon.strftime(DATETIME_FORMAT),
```

Comment:
ideally we'd actually have a `execute_server_cursor(query, params) -> Iterable[dict = row]` method that handles the dialect-specific cursor logic, and then the `get_rows` method can stay simple and only have the query + parameter business logic